src/status_im2/common/scan_qr_code/view.cljs

```diff
@@ -0,0 +1,243 @@
+(ns status-im2.common.scan-qr-code.view
+  (:require [clojure.string :as string]
+            [oops.core :as oops]
+            [quo2.core :as quo]
+            [quo2.foundations.colors :as colors]
+            [react-native.blur :as blur]
+            [react-native.camera-kit :as camera-kit]
+            [react-native.core :as rn]
+            [react-native.hole-view :as hole-view]
+            [react-native.permissions :as permissions]
+            [react-native.safe-area :as safe-area]
+            [reagent.core :as reagent]
+            [status-im2.common.device-permissions :as device-permissions]
+            [status-im2.common.scan-qr-code.style :as style]
+            [utils.debounce :as debounce]
+            [utils.i18n :as i18n]
+            [utils.re-frame :as rf]
+            [utils.transforms :as transforms]
+            [quo2.theme :as quo.theme]))
+
+(defonce camera-permission-granted? (reagent/atom false))
```

Comment:
Not for this PR. We need to move the permission states to `app-db` to prevent duplication.